### PR TITLE
Enable DataRow/DataRowView adapters in PowerShell Core

### LINKED ISSUE #4232: PowerShell Core - System.Data DataRow not providing the correct result.
I didn't pay too much attention in Beta 2 but after upgrading to Beta 3 I'm concern.

Steps to reproduce
------------------
Run Sample Code:
```powershell
## - Connect and run T-SQL query
$Global:getSvr = "mtrinidadlt2,1437";
$Global:con = "server=$Global:getSvr;database=Master;Integrated Security=false;User ID=sa;" + 'Password=$Adm1n!';
$SqlQry = @'
sp_who2
'@;

$sda = New-Object System.Data.SqlClient.SqlDataAdapter ($SqlQry, $global:con)
$sdt = New-Object System.Data.DataTable
$sda.fill($sdt) | Out-Null

$sdt.rows | Select -first 5 | Format-Table
```

Expected behavior
-----------------
DataRow data to provide proper objects results column with its data value. It works in Windows PowerShell.
(See Image)
pswindatarowok_2017-07-12_13-55-58

Actual behavior
---------------
Seems like a Python list array result: (See image)
pscoredatarowissue_2017-07-12_13-55-58

Environment data
----------------
Both OS's:
Windows 10 Build 16237
Ubuntu 16.04.02

```powershell
PS C:\Program Files\PowerShell\6.0.0-beta.3> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
GitCommitId                    v6.0.0-beta.3
OS                             Microsoft Windows 10.0.16237
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

### PR DESCRIPTION
**Pull request by daxian-dbw**
Fix #4232 

Enable the DataRow/DataRowView adapters in PowerShell Core by moving the implementation from `ExtraAdapter.cs` to `CoreAdapter.cs`.
Add basic tests too.

### CODE REVIEW

File: src/System.Management.Automation/engine/CoreAdapter.cs [hunk 1]
```diff
@@ -5007,6 +5008,221 @@ private static XmlNode[] FindNodes(object obj, string propertyName, StringCompar
         }
     }
 
+    /// <summary>
+    /// Deals with DataRow objects
+    /// </summary>
+    internal class DataRowAdapter : PropertyOnlyAdapter
```
Comment (TravisEz13) [inline src/System.Management.Automation/engine/CoreAdapter.cs]: Why don't we follow the one class per file practice?

**Review by TravisEz13 — APPROVED** [review]

File: src/System.Management.Automation/engine/CoreAdapter.cs (hunk 1 above)
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/CoreAdapter.cs]: Core adapters have always been in the same file. Some of them were moved to `ExtraAdapter.cs` previously due to missing the corresponding type in CoreCLR. Now we are gradually moving them back.

**Comment by MaximoTrinidad** [comment]
Hi @daxian-dbw 

Any idea when can I test this change?  Looking forward to give it a shot.
:)

Thanks,

File: src/System.Management.Automation/engine/CoreAdapter.cs (hunk 1 above)
Comment (TravisEz13) [inline src/System.Management.Automation/engine/CoreAdapter.cs]: The question is more about the project in general.

**Comment by daxian-dbw** [comment]
@MaximoTrinidad The release of beta.5 will on 8/3. But you can build powershell locally to verify the fix.

**Pull request merged**